searx/search/checker/background.py

```diff
@@ -0,0 +1,106 @@
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+import json
+import random
+import time
+import threading
+import os
+import signal
+
+from searx import logger, settings, searx_debug
+from searx.exceptions import SearxSettingsException
+from searx.search.processors import processors
+from searx.search.checker import Checker
+from searx.shared import schedule, storage
+
+
+CHECKER_RESULT = 'CHECKER_RESULT'
+running = threading.Lock()
+
+
+def _get_interval(every, error_msg):
+    if isinstance(every, int):
+        every = (every, every)
+    if not isinstance(every, (tuple, list))\
+       or len(every) != 2\
+       or not isinstance(every[0], int)\
+       or not isinstance(every[1], int):
+        raise SearxSettingsException(error_msg, None)
+    return every
+
+
+def _get_every():
+    every = settings.get('checker', {}).get('scheduling', {}).get('every', (300, 1800))
+    return _get_interval(every, 'checker.scheduling.every is not a int or list')
+
+
+def get_result():
+    serialized_result = storage.get_str('CHECKER_RESULT')
+    if serialized_result is not None:
+        return json.loads(serialized_result)
+
+
+def run():
+    if not running.acquire(blocking=False):
+        return
+    try:
+        logger.info('Starting checker')
+        result = {}
+        for name, processor in processors.items():
+            logger.debug('Checking %s engine', name)
+            checker = Checker(processor)
+            checker.run()
+            if checker.test_results.succesfull:
+                result[name] = {'status': True}
```

Comment:
Nit: maybe instead of `status` we could name it `success`, so the boolean value makes more sense.

Reply:
Fixed in this commit https://github.com/searx/searx/pull/2419/commits/a7e980251b88e12167dd2012aa1e36c76657e6bf

Also answer @HLFH  question : 

> Does the command line searx-checker status exist? So I can check if it is enabled or disabled?

If the checker is disabled, `/stats/checker` returns:
```json
{
  "status": "disabled",
  "timestamp": 1610388000
}
```

* `timestamp` is round to hour. Basically here it shows when the instance was started. Is it an issue ? It can be rounded to day if necessary.
* `kill -SIGUSR1 ...` can still trigger the checker.

The status is:
* `ok` after an update.
* `error` if an exception stop the checker